src/abstractdataframe/show.jl

```diff
@@ -458,103 +503,24 @@ function Base.show(io::IO,
-#' @description
-#'
-#' Render an AbstractDataFrame to stdout with or without chunking. See
-#' other `show` documentation for details. This is mainly used to force
-#' showing the AbstractDataFrame in chunks.
-#'
-#' @param df::AbstractDataFrame An AbstractDataFrame.
-#' @param allcols::Bool Should only a subset of columns that fits
-#'        the device width be printed? Defaults to `false`.
-#'
-#' @returns `nothing` value.
-#'
-#' @examples
-#'
-#' df = DataFrame(A = 1:3, B = ["x", "y", "z"])
-#' show(df, true)
-function Base.show(df::AbstractDataFrame,
-                   allcols::Bool = false) # -> Nothing
-    return show(stdout, df, allcols)
-end
-
-#' @exported
-#' @description
-#'
-#' Render all of the rows of an AbstractDataFrame to an IO system. See
-#' `show` documentation for details.
-#'
-#' @param io::IO The IO system to which `df` will be printed.
-#' @param df::AbstractDataFrame An AbstractDataFrame.
-#' @param allcols::Bool Should only a subset of columns that fits
-#'        the device width be printed? Defaults to `true`.
-#' @param rowlabel::Symbol What label should be printed when rendering the
-#'        numeric ID's of each row? Defaults to `"Row"`.
-#' @param displaysummary::Bool Should a brief string summary of the
-#'        AbstractDataFrame be rendered to the IO system before printing the
-#'        contents of the renderable rows? Defaults to `true`.
-#'
-#' @returns `nothing` value.
-#'
-#' @examples
-#'
-#' df = DataFrame(A = 1:3, B = ["x", "y", "z"])
-#' showall(stdout, df, false, :Row, true)
-function showall(io::IO,
-                 df::AbstractDataFrame,
-                 allcols::Bool = true,
-                 rowlabel::Symbol = :Row,
-                 displaysummary::Bool = true) # -> Nothing
-    rowindices1 = 1:size(df, 1)
-    rowindices2 = 1:0
-    maxwidths = getmaxwidths(df, rowindices1, rowindices2, rowlabel)
-    width = getprintedwidth(maxwidths)
-    showrows(io,
-             df,
-             rowindices1,
-             rowindices2,
-             maxwidths,
-             !allcols,
-             allcols,
-             rowlabel,
-             displaysummary)
-    return
-end
-
-#' @exported
-#' @description
-#'
-#' Render all of the rows of an AbstractDataFrame to stdout. See
-#' `showall` documentation for details.
-#'
-#' @param df::AbstractDataFrame An AbstractDataFrame.
-#' @param allcols::Bool Should only a subset of columns that fits
-#'        the device width be printed? Defaults to `true`.
-#'
-#' @returns `nothing` value.
-#'
-#' @examples
-#'
-#' df = DataFrame(A = 1:3, B = ["x", "y", "z"])
-#' showall(df, true)
-function showall(df::AbstractDataFrame,
-                 allcols::Bool = true) # -> Nothing
-    showall(stdout, df, allcols)
-    return
-end
-
-showall(io::IO, dfvec::AbstractVector{T}) where {T <: AbstractDataFrame} = foreach(df->showall(io, df), dfvec)
\ No newline at end of file
+function Base.show(df::AbstractDataFrame;
+                   allrows::Bool = false,
```

Comment:
This line and line below does not follow `:limit` contract of `show` with `io` parameter. If it is intentional then it should be documented (I can see the logic behind it as `stdout` typically will hve `:limit`, but probably in corner cases it can be redirected not to have a limit). If not I would make it consistent.

Reply:
Yeah, I had assumed `:limit` would always be true for `stdout`, but maybe that's not the case. It doesn't hurt to use `!get(stdout, :limit, false)` as for the more general method.

Reply:
Yeah AFAIK it should always work, and the worst that can happen is that we get the default value.

Reply:
OK, I see what you meant now: `!get(stdout, :limit, false)` always returns `true`, but we want `false` by default. I've pushed a commit to use `!get(stdout, :limit, true)`.